storage/excel/excel.go

```diff
@@ -0,0 +1,244 @@
+package excel
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+	"unicode"
+
+	"github.com/360EntSecGroup-Skylar/excelize"
+	"github.com/cube2222/octosql"
+	"github.com/cube2222/octosql/execution"
+	"github.com/pkg/errors"
+)
+
+var excelInitialDate = time.Date(1900, time.January, 1, 0, 0, 0, 0, time.UTC)
+
+const hoursInDay = 24
+const secondsInHour = 3600
+const nanosecondsInSecond = 1000000000
+
+func isCellNameValid(cell string) bool {
+	cellNameRegexp := "^[A-Z]+[1-9][0-9]*$"
+
+	r, _ := regexp.Compile(cellNameRegexp)
+
+	return r.MatchString(cell)
+}
+
+func getRowAndColumnFromCell(cell string) (row, col string) {
+	for index, char := range cell {
+		if unicode.IsDigit(char) {
+			return cell[index:], cell[:index]
+		}
+	}
+
+	return "", cell
+}
+
+func getCoordinatesFromCell(cell string) (row int, col int, err error) {
+	rowStr, colStr := getRowAndColumnFromCell(cell)
+	col = excelize.TitleToNumber(colStr)
+
+	rowTmp, err := strconv.ParseInt(rowStr, 10, 64)
+	if err != nil {
+		return 0, 0, errors.Wrap(err, "couldn't parse row")
+	}
+
+	return int(rowTmp) - 1, col, nil
+}
+
+func isAllZs(cell string) bool {
+	for _, char := range cell {
+		if char != 'Z' {
+			return false
+		}
+	}
+
+	return true
+}
+
+func getNextColumn(cell string) string {
+	if isAllZs(cell) {
+		return strings.Repeat("A", len(cell)+1)
+	}
+
+	nextCol := ""
+
+	index := len(cell)
+	for i := len(cell) - 1; i >= 0; i-- {
+		index--
+
+		if cell[i] == 'Z' {
+			nextCol += "A"
+		} else {
+			nextCol = string(cell[i]+1) + nextCol
+			break
+		}
+	}
+
+	return cell[:index] + nextCol
+}
+
+func createCellName(colName string, row int) string {
+	return colName + strconv.Itoa(row)
+}
+
+func isDateStyle(style int) bool {
+	return style == 2 || style == 3 || style == 4
+}
+
+func getDateFromExcelTime(timeValueStr string) (time.Time, error) {
+	timeValue, err := strconv.ParseFloat(timeValueStr, 64)
+
+	if err != nil {
+		return time.Now(), errors.Wrap(err, "couldn't convert string to float")
+	}
+
+	daysPart := int64(timeValue)
+	addedDays := excelInitialDate.Add(time.Hour * 24 * time.Duration(daysPart-2))
+
+	timeLeft := timeValue - float64(daysPart) //fractions of day
+	timeLeft *= nanosecondsInSecond * secondsInHour * hoursInDay
+
+	fullDate := addedDays.Add(time.Nanosecond * time.Duration(timeLeft))
+
+	return fullDate, nil
+}
+
+func padRow(row []string, targetLen int) []string {
+	difference := targetLen - len(row)
+
+	for i := 0; i < difference; i++ {
+		row = append(row, "")
+	}
+
+	return row
+}
+
+func (rs *RecordStream) parseDataTypes(row []string) error {
```

Comment:
In my opinion this shouldn't work in place and should return a []octosql.Value. This way you could already use ParseType on fields that need it, and also do your custom time parsing here, only putting those values through normalizeValue